### ISSUE TITLE
Not working on Real World Image

### ISSUE THREAD

**Issue by tulasiram58827** [zzxvictor/License-super-resolution#1]
I tried this network on real world image but it didn't worked. I am attaching the [gist](https://colab.research.google.com/drive/17x03Y2VlpN92GFQKFkhpX1tmM2_WNbJb?usp=sharing) and link to image also attached in the colab notebook itself. Did I missed something or model failed?

**Comment by zzxvictor** [comment]
we noticed this problem at the end of the semester. We figured out that it is very likely that our data preprocessing script needs some revision. In order to create images that look like pictures taken far away by a low-quality camera, it seems that bilinear downsampling isn't enough. Unfortunately we didn't have time to fix this and I am 120% occupied by internship work. Here's my suggestion: retrain the pretrained model on images generated with the follwoing strategy.

What our data processing script does:  
* downsample by 8 
* add random noise 
* change constrast and brightness randomly

A potential better way to do this is:  
* *convolve* the high resolution image with a *gaussian* filter (hyparameter needs tuning)
* downsample by 8 
* add random noise 
* change constrast and brightness randomly

Check out the post dedicated to this discussion. [reference](https://dsp.stackexchange.com/questions/52721/why-should-an-image-be-blurred-using-a-gaussian-kernel-before-downsampling)

Cheers

**Comment by tulasiram58827** [comment]
Thanks for the info. Just a small doubt it seems like your model is working only on the images you trained. Doesn't it overfitted somehow?

**Comment by zzxvictor** [comment]
I think this issue is not about overfitting as I only showed the results on the hold-out test set. It probably because the input data, although downsampled and pixelated, is different from a low-resolution image in the real world and the transformation learned by the model doesn't apply.

**Comment by tulasiram58827** [comment]
Okay. Let me retrain with the suggested procedure and will get back to you

**Comment by zzxvictor** [comment]
That would be wonderful! Just add Gaussian blurring to the process() function in the DataLoader class. My advice is to start with a smaller generated to obtain ok results (classic supervised model) and then move on to GANs to achieve better results. 
Also, don't 100% rely on MSE loss as they are not a good metric. Use the content loss function instead.

**Comment by tulasiram58827** [comment]
Yeah sure thanks for advice. Will ask for suggestions through issues if I stuck with anything.

**Comment by zzcqinag** [comment]
hi! how to add Gaussian blurring in the process()? @ zzxvictor

**Comment by zzxvictor** [comment]
You can add it to the data processing pipeline. Here's the documentation for Gaussian Blurring by OpenCV: https://docs.opencv.org/master/d4/d13/tutorial_py_filtering.html

**Comment by alevangel** [comment]
There are any news on this aspect? @tulasiram58827

**Comment by siddagra** [comment]
It seems like it is definitely over-fitting because there is no way a model can extract characters from a jumbeled mess of pixels that even no human could... It's a GAN, not magic...
Most likely there must have been some train/test leakage.
